### PR TITLE
chore: make slot checks boolean

### PR DESCRIPTION
**Pull request by lee-chase**
Change slot checks to boolean values so they show as true/false during debug/testing.

### CODE REVIEW

File: packages/core/src/components/cv-data-table/cv-data-table-row.vue [hunk 1]
```diff
@@ -51,11 +51,11 @@ export default {
   },
   mounted() {
     // NOTE: this.$slots is not reactive so needs to be managed on beforeUpdate
-    this.dataExpandable = this.$slots.expandedContent !== undefined;
+    this.dataExpandable = !!(this.$slots.expandedContent !== undefined);
```
Comment (ZrianinaMariia) [inline packages/core/src/components/cv-data-table/cv-data-table-row.vue]: I guess here it is already Boolean, so no need for `!!` or?

File: packages/core/src/components/cv-data-table/cv-data-table-row.vue [hunk 2]
```diff
@@ -51,11 +51,11 @@ export default {
...
     this.$parent.$emit('cv:mounted', this);
   },
   updated() {
-    this.dataExpandable = this.$slots.expandedContent !== undefined;
+    this.dataExpandable = !!(this.$slots.expandedContent !== undefined);
```
Comment (ZrianinaMariia) [inline packages/core/src/components/cv-data-table/cv-data-table-row.vue]: I guess here it is already Boolean, so no need for !! or?

File: packages/core/src/components/cv-ui-shell/cv-header.vue [hunk 3]
```diff
@@ -31,10 +31,10 @@ export default {
   },
   mounted() {
     // NOTE: this.$slots is not reactive so needs to be managed on beforeUpdate
-    this.hasGlobalHeader = this.$slots['header-global'] !== undefined;
+    this.hasGlobalHeader = !!(this.$slots['header-global'] !== undefined);
```
Comment (ZrianinaMariia) [inline packages/core/src/components/cv-ui-shell/cv-header.vue]: Also not sure if `!!` is needed here.

File: packages/core/src/components/cv-ui-shell/cv-side-nav-link.vue [hunk 4]
```diff
@@ -37,7 +37,7 @@ export default {
     };
   },
   beforeUpdate() {
-    this.hasNavIcon = this.$slots['nav-icon'] !== undefined;
+    this.hasNavIcon = !!(this.$slots['nav-icon'] !== undefined);
```
Comment (ZrianinaMariia) [inline packages/core/src/components/cv-ui-shell/cv-side-nav-link.vue]: Not sure if `!!` is needed here.

File: packages/core/src/components/cv-data-table/cv-data-table.vue [hunk 5]
```diff
@@ -324,10 +324,10 @@ export default {
   methods: {
     checkSlots() {
       // NOTE: this.$slots is not reactive so needs to be managed on beforeUpdate
-      this.hasBatchActions = this.$slots['batch-actions'];
-      this.hasActions = this.$slots.actions;
-      this.hasToolbar = this.$slots.actions || this.$listeners.search || this.$slots['batch-actions'];
-      this.isHelper = this.$slots['helper-text'] !== undefined || (this.helperText && this.helperText.length);
+      this.hasBatchActions = !!this.$slots['batch-actions'];
```
Comment (ZrianinaMariia) [inline packages/core/src/components/cv-data-table/cv-data-table.vue]: In other places, e.g. `cv-side-nav-menu`, you use comparison with undefined `this.$slots['nav-icon'] !== undefined`. I think it makes sense to have it consistent.

File: packages/core/src/components/cv-data-table/cv-data-table-row.vue (hunk 1 above)
Comment (lee-chase) [reply packages/core/src/components/cv-data-table/cv-data-table-row.vue]: Good point, that is a danger of find and replace. Will correct.

**Pull request merged**